### PR TITLE
Create new Sort command

### LINKED ISSUE #118: Create new Sort command

### PR DESCRIPTION
**Pull request by seah-minlong**
Fixes #118 

List and sort functionality are combined.

Separating them allows for sort to be utilised after any other command.

Let's
* Create SortCommand and SortCommandParser class
* Remove SortOption from ListCommand
* Update ListCommand to ignore any preamble or extra parameters

### CODE REVIEW

**Review by Justin-Yeo — APPROVED** [review]
LGTM! I like the idea of keeping the code easy for expansion for future sorting options 👍

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (sortOption == null) {
+            model.clearPersonSort();
+            return new CommandResult(MESSAGE_DEFAULT_SUCCESS);
+        }
+
+        switch (sortOption.toString()) {
+        case SortOption.SORT_NAME:
```
Comment (Justin-Yeo) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:47]: for this are we able to import at the top, then simply write SORT_NAME

**Review by Justin-Yeo — CHANGES_REQUESTED** [review]
minor code quality improvement

File: src/main/java/seedu/address/logic/commands/SortCommand.java (hunk 1 above)
Comment (seah-minlong) [reply src/main/java/seedu/address/logic/commands/SortCommand.java:47]: Sure that sounds great

**Pull request merged**